### PR TITLE
Fix explain plan formatting in sqllogictest

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/6328 (noticed by @mustafasrepo  on https://github.com/apache/arrow-datafusion/pull/6234)

# Rationale for this change

For some reason, `cargo test --test sqllogictests -- --complete` when completing tests strips the leading whitespace sometimes.  Then the comparison also ignores leading whitespace

So sometimes the explain plans captured by `--compete` would take a plan like:
```
Projection 
  Sort
    Window
     Sort
```

And display it like:

```
Projection 
  Sort
    Window
   Sort
```

# What changes are included in this PR?

1. ~Update the normalization code in the DataFusion `sqlogictest` to prefix explain plan lines with `|` ~
2. Update the normalization code in the DataFusion `sqlogictest` to replace leading spaces with `-`
2. Recapture  the explain plans `cargo test --test sqllogictests -- --complete`

# Are these changes tested?
Yes

# Are there any user-facing changes?
no

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/window.slt [hunk 1]
```diff
@@ -1348,18 +1348,18 @@ EXPLAIN SELECT
...
+|        TableScan: aggregate_test_100 projection=[c9]
 physical_plan
-ProjectionExec: expr=[c9@0 as c9, FIRST_VALUE(aggregate_test_100.c9) ORDER BY [aggregate_test_100.c9 ASC NULLS LAST] ROWS BETWEEN 1 PRECEDING AND 5 FOLLOWING@4 as fv1, FIRST_VALUE(aggregate_test_100.c9) ORDER BY [aggregate_test_100.c9 DESC NULLS FIRST] ROWS BETWEEN 1 PRECEDING AND 5 FOLLOWING@1 as fv2, LAG(aggregate_test_100.c9,Int64(2),Int64(10101)) ORDER BY [aggregate_test_100.c9 ASC NULLS LAST] RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW@5 as lag1, LAG(aggregate_test_100.c9,Int64(2),Int64(10101)) ORDER BY [aggregate_test_100.c9 DESC NULLS FIRST] ROWS BETWEEN 10 PRECEDING AND 1 FOLLOWING@2 as lag2, LEAD(aggregate_test_100.c9,Int64(2),Int64(10101)) ORDER BY [aggregate_test_100.c9 ASC NULLS LAST] RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW@6 as lead1, LEAD(aggregate_test_100.c9,Int64(2),Int64(10101)) ORDER BY [aggregate_test_100.c9 DESC NULLS FIRST] ROWS BETWEEN 10 PRECEDING AND 1 FOLLOWING@3 as lead2]
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/window.slt:1357]: Here is an example I think that shows the indent being properly displayed

**Review by mustafasrepo — APPROVED** [review]

**Comment by mustafasrepo** [comment]
I have reviewed the PR. This PR is LGTM!. Now the final plan is more clear, especially when the query involves executors with multiple children (such as union, join, etc.).  @alamb thanks for this improvement.

**Comment by alamb** [comment]
> During experimenting, I have tried to mess indentation and run the tests. Tests still pass. For instance both

Hmm, this seems pretty bad as if the plan changes the tests won't fail. Let me see if I can find something that looks better

**Comment by alamb** [comment]
@mustafasrepo I updated the code to replace leading spaces with `-` rather than prefixing with `|`

Now when I mess up the indent (remove a `-` the tests fail as I expect.

What do you think

**Comment by mustafasrepo** [comment]
> @mustafasrepo I updated the code to replace leading spaces with `-` rather than prefixing with `|`
> 
> Now when I mess up the indent (remove a `-` the tests fail as I expect.
> 
> What do you think?

This is great!. Now we can make sure that indentation level is exact. Thanks for this fix.

**Review by comphead — COMMENTED** [review]
Not related for this PR but rather than for sqllogic improvement. 
Sqllogic tests doesn't cover cases if the column name is not expected.
The same relates to datatype, but this can be solved by extra `arrow_typeof`

**Comment by alamb** [comment]
> Sqllogic tests doesn't cover cases if the column name is not expected.

Yes, I agree this is not currently covered by the sqllogictest -- I'll file a ticket to try to work on this. 

> The same relates to datatype, but this can be solved by extra arrow_typeof

SQLlogictests does verify the (sql) types of the output (or example, the `II` in the following means the columns should be integers). It is not as fine grained as the arrow type system, but it is better than nothing

```
query II
SELECT * FROM foo_schema.bar;
----
1 2
```

**Pull request merged**